### PR TITLE
INT B-18386 Enhancements

### PR DESCRIPTION
**Pull request by danieljordan-caci**
Added enhancements to allow for testing & conditional rendering for previous PR of [B-18309](https://github.com/transcom/mymove/pull/11811)

This PR does the following:
- added conditional rendering if the data came back as an empty array
- added in param use to allow for testing - a user can navigate to `/moves` and add the following params and have different test data populated:
`?moveData=PCS`
`?moveData=retirement`
`?moveData=separation`
`?moveData=noPreviousMoves`
`?moveData=noCurrentMove`
`?moveData=noMoves`

### CODE REVIEW

File: src/pages/MyMove/Multi-Moves/MultiMovesLandingPage.jsx [hunk 1]
```diff
@@ -18,6 +25,35 @@ import Helper from 'components/Customer/Home/Helper';
...
+      moves = mockMovesNoCurrentMoveWithPreviousMoves;
+      break;
+    case 'noMoves':
+      moves = mockMovesNoCurrentOrPreviousMoves;
+      break;
+    default:
+      moves = mockMovesPCS;
+      break;
+  }
+  // ! end of test data
```
Comment (cameroncaci) [inline src/pages/MyMove/Multi-Moves/MultiMovesLandingPage.jsx:56]: This is great for manual testers and reviewing visual mockups, however given a month or two I can absolutely see this manual method being forgotten about. Could we introduce some automated tests such as Jest?

I'm tracking that Playwright isn't quite possible due to the lack of multi move functionality.

I see that this is marked as an enhancement as well, but I don't see anything detailed in the ticket description. Let me know if the manual method was intentionally pursued as a temporary implementation.

**Review by cameroncaci — CHANGES_REQUESTED** [review]

File: src/pages/MyMove/Multi-Moves/MultiMovesLandingPage.jsx (hunk 1 above)
Comment (danieljordan-caci) [reply src/pages/MyMove/Multi-Moves/MultiMovesLandingPage.jsx:56]: Yes this is just meant for temporary testing and will be removed once additional backlog items are completed (specifically [B-18314](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A873920&RoomContext=TeamRoom%3A808731&concept=TeamRoom). In that Agility ticket, I put in the description to remove this functionality.

Comment (danieljordan-caci) [reply src/pages/MyMove/Multi-Moves/MultiMovesLandingPage.jsx:56]: Also - I got BL items confused. This BL item is an enhancement for [this PR](https://github.com/transcom/mymove/pull/11811)

Comment (cameroncaci) [reply src/pages/MyMove/Multi-Moves/MultiMovesLandingPage.jsx:56]: Gotcha, thanks for the clarification

**Review by cameroncaci — APPROVED** [review]

**Pull request merged**